Pangolin/LogViewer/Pages/ApplicationSettings.razor

```diff
@@ -0,0 +1,119 @@
+﻿@page "/applicationsettings"
+@attribute [Authorize]
+
+@using EnderPi.Framework.DataAccess;
+@using Microsoft.Extensions.Configuration
+@using EnderPi.Framework.Pocos;
+@inject IConfiguration Configuration
+@inject EnderPi.Framework.Logging.Logger logger
+@inject ConfigurationDataAccess dataAccess
+
+<h3>Application Settings</h3>
+
+<EditForm Model="SettingModel">
+    <table>
+        <tr>
+            <td>Application Name:</td>
+            <td><InputText @bind-Value="SettingModel.Application" /></td>
+        </tr>
+        <tr>
+            <td>Setting Name:</td>
+            <td><InputText @bind-Value="SettingModel.Name" /></td>
+        </tr>
+
+        <tr>
+            <td>Setting Value:</td>
+            <td><InputText @bind-Value="SettingModel.Value" /></td>
+        </tr>
+    </table>
+</EditForm>
+<p>
+    <button type="button" class="btn btn-primary" @onclick="AddSetting">Add New Setting</button>
+</p>
+
+
+<table class="table">
+    <thead>
+        <tr>
+            <th>Application Name</th>
+            <th>Setting Name</th>
+            <th>Setting Value</th>
+            <th>Delete Setting</th>
+        </tr>
+    </thead>
+    <tbody>
+
+        @for (int i = 0; i < _applicationSettings.Length; i++)
+        {
+            var setting = _applicationSettings[i];
+            int j = i;
+            <tr>
+                <td>@setting.ApplicationName</td>
+                <td>@setting.SettingName</td>
+                <td><input @bind-value="@setting.SettingValue" @onblur="@(()=>UpdateSettingValue(setting.ApplicationName, setting.SettingName, setting.SettingValue))" /></td>
+                <td><button class="btn btn-primary" @onclick="@(()=>DeleteSetting(setting.ApplicationName, setting.SettingName))">Delete</button></td>
+            </tr>
+        }
+    </tbody>
+</table>
+
+@code {
+    private ApplicationSetting[] _applicationSettings;
+
+    private LogViewer.Models.ApplicationSettingModel SettingModel = new Models.ApplicationSettingModel();
+
+    protected override Task OnInitializedAsync()
+    {
+        RefreshUI();
+        return base.OnInitializedAsync();
+    }
+
+    private void RefreshUI()
+    {
+        _applicationSettings = dataAccess.GetApplicationSettings();
+        SettingModel.Application = "";
+        SettingModel.Name = "";
+        SettingModel.Value = "";
+    }
+
+    public void AddSetting()
+    {
+        try
+        {
+            dataAccess.CreateApplicationSetting(SettingModel.Application, SettingModel.Name, SettingModel.Value);
+            RefreshUI();
+        }
+        catch (Exception ex)
+        {
+            logger.Log(ex.ToString(), EnderPi.Framework.Logging.LoggingLevel.Error);
```

Comment:
Add prompt to check the database for the error, else it will to appear to have worked. And try refreshing an additional time?